### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by mullermp**
RBS struct members
Refactor interceptor apply out of interceptor list
Rename yardType to docType
Extra RBS in hearth + some minor tweaks

other minor stuff.

### CODE REVIEW

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/RubySymbolProvider.java [hunk 1]
```diff
@@ -279,9 +279,9 @@ public Symbol timestampShape(TimestampShape shape) {
     @Override
     public Symbol listShape(ListShape shape) {
         Symbol member = toSymbol(model.expectShape(shape.getMember().getTarget()));
-        String rbsType = "Array[" + member.getProperty("rbsType").get() + "]";
-        String yardType = "Array<" + member.getProperty("yardType").get() + ">";
-        return createSymbolBuilder(shape, getDefaultShapeName(shape, "List__"), rbsType, yardType, moduleName)
+        String rbsType = "::Array[" + member.getProperty("rbsType").get() + "]";
```
Comment (alextwoods) [inline codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/RubySymbolProvider.java:282]: Out of scope of this PR - but I think related to my comment from previous PR - I think maybe we shoudl define a RubyType class and create instances for things like Array<Type>. that will know how to do the doc types, rbs, contarints, ect....  I think this applies here and in the middleware and possibly in other places as well.

File: hearth/lib/hearth/middleware/build.rb [hunk 2]
```diff
@@ -18,7 +18,8 @@ def initialize(app, builder:)
       # @param context
       # @return [Output]
       def call(input, context)
-        interceptor_error = context.interceptors.apply(
+        interceptor_error = Interceptor.apply(
+          interceptors: context.interceptors,
```
Comment (alextwoods) [inline hearth/lib/hearth/middleware/build.rb:22]: Will the interceptors we pass ever differ from whats on context?  I think they never would.  Should we drop this parameter from the method and just use the interceptors that are on the context internally?

**Review by alextwoods — APPROVED** [review]
LGTM - just minor comments

File: hearth/lib/hearth/middleware/build.rb (hunk 2 above)
Comment (mullermp) [reply hearth/lib/hearth/middleware/build.rb:22]: Ah yeah I think we could.

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/RubySymbolProvider.java (hunk 1 above)
Comment (mullermp) [reply codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/RubySymbolProvider.java:282]: I'm not sure what that would look like... But yeah we can potentially do that outside this change.

**Pull request merged**